c7n/policy.py

```diff
@@ -80,7 +80,7 @@ def __init__(self, data, options):
             if 'all' in options.regions:
                 options.regions = all_regions
             for region in options.regions:
-                if region not in all_regions:
+                if len(all_regions) > 0 and region not in all_regions:
```

Comment:
I reverted this duct tape for this PR. Let me know if there are any changes you want in here.